src/main/java/com/synopsys/integration/alert/workflow/startup/component/StartupComponent.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * blackduck-alert
+ *
+ * Copyright (c) 2019 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.workflow.startup.component;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class StartupComponent {
+    private final Logger logger = LoggerFactory.getLogger(StartupComponent.class);
+
+    public void initializeComponent() {
+        final String runningStartupComponentLog = String.format("Running startup component: %s", getClass().getSimpleName());
+        logger.info(runningStartupComponentLog);
+        run();
+    }
+
+    public abstract void run();
```

Comment:
can you make this protected since we want initializeComponent to be the public interface.  Also change the method from run to initialize.  Run makes me think of it as a thread which it is not at the moment.